### PR TITLE
libteeacl: condition libteeacl with WITH_TEEACL

### PR DESCRIPTION
**Pull request by etienne-lms**
Build and embed libteeacl upon WITH_TEEACL=1 (default configuration). This configuration switch allows one to build OP-TEE client without dependencies on pkg-config and libuuid  when OP-TEE ACL for PKCS11 is not needed:
 cmake -DWITH_TEEACL=0 ...
or
 make WITH_TEEACL=0 ...

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,7 +61,10 @@ check-libuuid:
 install: copy_export
 
 clean: clean-libteec clean-tee-supplicant clean-cscope clean-libckteec \
-	clean-libseteec clean-libteeacl
+	clean-libseteec
+ifeq ($(WITH_TEEACL),1)
+clean-libteeacl
```
Comment (jforissier) [inline Makefile:66]: `clean: clean-libteeacl`

File: CMakeLists.txt [hunk 2]
```diff
@@ -37,7 +38,9 @@ add_subdirectory (libteec)
 add_subdirectory (tee-supplicant)
 add_subdirectory (public)
 add_subdirectory (libckteec)
+if(WITH_TEEACL)
 find_package (PkgConfig REQUIRED)
 pkg_check_modules(uuid REQUIRED IMPORTED_TARGET uuid)
 add_subdirectory (libteeacl)
+endif (WITH_TEEACL)
```
Comment (jforissier) [inline CMakeLists.txt:45]: Prefer no space?

**Review by jforissier — COMMENTED** [review]
With the comments below addressed, LGTM.

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: CMakeLists.txt [hunk 3]
```diff
@@ -5,6 +5,7 @@ project (optee_client C)
 set (CMAKE_TOOLCHAIN_FILE CMakeToolchain.txt)
 
 set (CFG_WERROR 1 CACHE BOOL "Build with -Werror")
+set (WITH_TEEACL 1 CACHE BOOL "Build libteeacl")
```
Comment (eaaltonen) [inline CMakeLists.txt:8]: I think it's more typical in CMake to handle boolean options with `option`, so
```
option(CFG_WERROR "Build with -Werror" TRUE)
option(WITH_TEEACL "Build libteeacl" TRUE)
```

also CMake project's own style seems to not have space between function and `(`.

Comment (eaaltonen) [reply CMakeLists.txt:8]: There's teeacl also in `copy_export`, so should probably also add
```
@@ -165,11 +165,9 @@
 	cp libckteec/include/*.h $(DESTDIR)$(INCLUDEDIR)
 	cp -d ${O}/libckteec/libckteec.so* $(DESTDIR)$(LIBDIR)
 	cp -d ${O}/libckteec/libckteec.a $(DESTDIR)$(LIBDIR)
-ifeq ($(WITH_TEEACL),1)
 	cp libteeacl/include/*.h $(DESTDIR)$(INCLUDEDIR)
 	cp -d ${O}/libteeacl/libteeacl.so* $(DESTDIR)$(LIBDIR)
 	cp -d ${O}/libteeacl/libteeacl.a $(DESTDIR)$(LIBDIR)
-endif	
 	cp libseteec/include/*.h $(DESTDIR)$(INCLUDEDIR)
 	cp -d ${O}/libseteec/libseteec.so* $(DESTDIR)$(LIBDIR)
 	cp -d ${O}/libseteec/libseteec.a $(DESTDIR)$(LIBDIR)
```

**Review by etienne-lms — COMMENTED** [review]
updated with comments addressed, maybe to much: i've added some exptra space char but  in the end, i don't think it's really expected.

File: CMakeLists.txt (hunk 2 above)
Comment (eaaltonen) [reply CMakeLists.txt:45]: I'd prefer CMake files without space, but a style agreement in either way would be welcome.

**Comment by eaaltonen** [comment]
`Reviewed-by: Eero Aaltonen <eero.aaltonen@vaisala.com>`
`Tested-by` as well, if you use it (_built the 4 combinations and visually inspected the install tree_).

**Comment by etienne-lms** [comment]
Thanks @eaaltonen for the review and testing.
I'll get rid of the added space char. CMake litterature does not seem to recommend or suggest them.

> _built the 4 combinations ..._

You mean `CFG_WERROR`/`WITH_TEEACL` conbinaisions.

**Comment by etienne-lms** [comment]
I've squashed the fixed commit and removed the added space characters.
Review tags applied (I think still they apply with removed space chars).

**Comment by jforissier** [comment]
> > You mean CFG_WERROR/WITH_TEEACL conbinaisions ?
> 
> I actually meant the combinations of `echo {make,cmake},WITH_TEEACL={0,1}`.

I suggest adding a few lines to `.github/workflows/ci.yml` to make sure everything is tested

File: CMakeLists.txt [hunk 4]
```diff
@@ -37,7 +38,9 @@ add_subdirectory (libteec)
 add_subdirectory (tee-supplicant)
 add_subdirectory (public)
 add_subdirectory (libckteec)
+if(WITH_TEEACL)
 find_package (PkgConfig REQUIRED)
 pkg_check_modules(uuid REQUIRED IMPORTED_TARGET uuid)
 add_subdirectory (libteeacl)
```
Comment (eaaltonen) [inline CMakeLists.txt:44]: Just now realized that should probably also indent the block within the `if(...)`.

Comment (etienne-lms) [reply CMakeLists.txt]: addressed in fixup commit.

Comment (eaaltonen) [reply CMakeLists.txt]: :+1: feel free to squash.

Comment (etienne-lms) [reply CMakeLists.txt]: done, thanks.

**Comment by etienne-lms** [comment]
Addressed review comment. Tags are already applied.

**Pull request merged**